### ISSUE TITLE
Request to Disable Cancel Button During Payment Processing

### ISSUE THREAD

**Issue by johndoeblocks** [Adyen/adyen-react-native#271]
**Describe the Bug**
While making payment requests using the `@adyen/react-native` package, users are able to click the cancel button on the payment interface during payment processing. This can lead to unintended issues, such as cancelling a payment inadvertently. 

**To Reproduce**
1. Initiate a payment request.
2. While the payment request is processing, click the cancel button.

**Expected Behavior**
Users should not be able to click the cancel button while a payment request is being processed. This prevents accidental cancellation of payments and ensures a smooth payment experience.

**Request to Disable Cancel Clicks During Payment Requests**
I kindly request the addition of a feature that allows the disabling of cancel clicks on the payment interface while payment requests are being processed. This enhancement would greatly improve the user experience by preventing any interruptions during critical payment transactions.

**Environment**
- Platform: [iOS]
- `@adyen/react-native` Package Version: [1.0.0]
- Device: [e.g. iPhone 14]
- OS: [iOS 16.1]

Is it possible to manually control the loading state of the drop in?

**Comment by descorp** [comment]
Hey @joaooliveiramanteigas 

Could you describe your case in more details?
Any specific payment method?

**Comment by johndoeblocks** [comment]
Hi @descorp ,

Yes, for instance Vipps payment. When coming back from Vipps we get redirect back to our app to perform the onAdditionalDetails and if the request takes some more time to complete, no loading UI is presented and user can eventually click the cancel button.

**Comment by descorp** [comment]
Hey @joaooliveiramanteigas 

Make sense. 
Thank you for feedback, will investigate!

**Comment by johndoeblocks** [comment]
This only seems to happen on iOS, verified today. Android keeps the loading spinner.

**Comment by johndoeblocks** [comment]
Do we have new information on this issue?

**Comment by descorp** [comment]
Hey @joaooliveiramanteigas 

iOS and Android redirect UI acts differently due to OS specific behaviour.

If [payments/details](https://docs.adyen.com/api-explorer/Checkout/70/post/payments/details) take too long, one possible option:

1) on `onAdditionalDetails` dismiss Payment UI with `hide(true)`
2) present your own spinner overlay
3) perform `payments/details` call.

**Comment by johndoeblocks** [comment]
Hello @descorp,

I've included video examples for your reference. In these videos, I've added a delay in the onAdditionalDetails function to simulate longer request times. Additionally, this issue is more to point out the different behaviors between iOS and Android. On iOS, users have the freedom to click the cancel button at any time, whereas on Android, the entire UI is locked with a loading spinner until the request is completed.

I appreciate your feedback and will definitely give your suggested approach a try. Thank you!

https://github.com/Adyen/adyen-react-native/assets/85570868/45bf76ac-9481-414c-b947-352e9c6286a3
https://github.com/Adyen/adyen-react-native/assets/85570868/2c7ae613-215b-4b23-a30d-a89177234f47
